app/services/billable_metrics/destroy_service.rb

```diff
@@ -23,6 +23,7 @@ def call
         .where(billable_metrics: {id: metric.id}).distinct.pluck(:id)
 
       ActiveRecord::Base.transaction do
+        metric.alerts.discard_all!
```

Comment:
I recently discovered the hard way that `discard_all` is loading every models in memory and perform validation/callbacks.
I guess in this case we want to use `update_all(deleted_at: Time.current)` instead

For reference: https://github.com/jhawthorn/discard?tab=readme-ov-file#performance-tuning

Reply:
Interestingly enough, we discovered this in the exact same transaction, 2 lines below 😅  #3747